client/blockbook/client.go

```diff
@@ -22,21 +20,62 @@ import (
 	"github.com/OpenBazaar/golang-socketio/protocol"
 	"github.com/OpenBazaar/multiwallet/client/transport"
 	"github.com/OpenBazaar/multiwallet/model"
+	"github.com/btcsuite/btcd/chaincfg"
 	"github.com/btcsuite/btcd/chaincfg/chainhash"
 	"github.com/btcsuite/btcutil"
+	"github.com/cpacia/bchutil"
 	"github.com/op/go-logging"
 	"golang.org/x/net/proxy"
 )
 
 var Log = logging.MustGetLogger("client")
 
+type wsWatchdog struct {
+	client    *BlockBookClient
+	done      chan struct{}
+	wsStopped chan struct{}
+}
+
+func newWebsocketWatchdog(client *BlockBookClient) *wsWatchdog {
+	return &wsWatchdog{
+		client:    client,
+		done:      make(chan struct{}, 0),
+		wsStopped: make(chan struct{}, 0),
+	}
+}
+
+func (w *wsWatchdog) guardWebsocket() {
+	for {
+		select {
+		case <-w.wsStopped:
+			time.Sleep(1 * time.Second)
+			Log.Warningf("websocket restarting")
+			w.client.SocketClient.Close()
```

Comment:
Shouldn't these `w.client.SocketClient` manipulations be guarded with the `listenLock` lock in the client? afaict that lock exists to protect the `SocketClient`. It should probably be renamed `socketClientMu` to denote what it's locking.